disnake/ext/commands/params.py

```diff
@@ -1053,7 +1059,8 @@ def expand_params(command: AnySlashCommand) -> List[Option]:
     Returns the created options
     """
     sig = signature(command.callback)
-    _, inter_param, params, injections = collect_params(command.callback)
+    # pass `sig` down to avoid having to call `signature(func)` another time
```

Comment:
```suggestion
    # pass `sig` down to avoid having to call `signature(func)` another time 
    # which may cause side effects with deferred annotations and warnings
```